aws/sdk/integration-tests/s3/tests/query-strings-are-correctly-encoded.rs

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_http::user_agent::AwsUserAgent;
+use aws_hyper::AwsMiddleware;
+use aws_sdk_s3::operation::ListObjectsV2;
+use aws_sdk_s3::{Credentials, Region};
+use aws_smithy_client::test_connection::capture_request;
+use aws_smithy_client::Client as CoreClient;
+use std::time::{Duration, UNIX_EPOCH};
+
+pub type Client<C> = CoreClient<C, AwsMiddleware>;
+
```

Comment:
Seems to me like these kinds of tests should be in `aws-sigv4`

Reply:
I can move it there, I only put it in S3 because that's where the user originally encountered the issue

Reply:
IMO, both is good—It's nice to test S3 directly so that we have a test that covers end-to-end codegen and wiring but a specific test is also not a bad idea